query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -111,6 +112,9 @@ public void cacheStarting(ComponentRegistry cr, Configuration cfg, String cacheN
    }
 
    private void addCacheDependencyIfNeeded(String cacheStarting, EmbeddedCacheManager cacheManager, Properties properties) {
+      if (!ClusterRegistryImpl.GLOBAL_REGISTRY_CACHE_NAME.equals(cacheStarting)) {
+         cacheManager.addCacheDependency(cacheStarting, ClusterRegistryImpl.GLOBAL_REGISTRY_CACHE_NAME);
+      }
```

Comment:
The ```addCacheDependency``` will enforce order only during cache stop, but caches are started in parallel.
Are the random failures caused by stopping the registry before the indexed/data caches?